frappe/desk/reportview.py

```diff
@@ -493,7 +494,11 @@ def delete_bulk(doctype, items):
 		except Exception:
 			# rollback if any record failed to delete
 			# if not rollbacked, queries get committed on after_request method in app.py
+			undeleted_items.append(d)
 			frappe.db.rollback()
+	if undeleted_items and len(items) != len(undeleted_items):
```

Comment:
I think it is necessary to provide protection against jamming. For some reason, the "undeleted_items" list may not always be empty. For example, if there is some error when deleting a document that is not related to dependent !

Reply:
If all the items in the list gives an error, the length of items and undeleted_items will be same then it will come out of the loop

Reply:
> If all items gives an error length will be same

Logically !
Didn't notice right away...

Reply:
Python will block when it recognizes recursion of 1000 level deep :smile:

Reply:
But I agree, let's limit retries to 5 levels or so and break it after that.

Reply:
Failures will be displayed I am only clearing once
When loop ends the actual failures will be displayed

Reply:
How can we apply a limit? a global flag?